dcrec/secp256k1/pubkey.go

```diff
@@ -5,72 +5,87 @@
+	// section 2.3.4 of [SEC1](https://secg.org/sec1-v2.pdf#subsubsection.2.3.4).
+	PubKeyFormatCompressedOdd byte = 0x03
+
+	// PubKeyFormatUncompressed is the identifier prefix byte for a public key
+	// when serialized according in the uncompressed format per section 2.3.3 of
+	// [SEC1](https://secg.org/sec1-v2.pdf#subsubsection.2.3.3).
+	PubKeyFormatUncompressed byte = 0x04
+
+	// PubKeyFormatHybridEven is the identifier prefix byte for a public key
+	// whose Y coordinate is even when serialized according to the hybrid format
+	// per section 4.3.6 of [ANSI X9.62-1998].
+	//
+	// NOTE: This format makes little sense in practice an therefore this
+	// package will not produce public keys serialized in this format.  However,
+	// it will parse them since they exist in the wild.
+	PubKeyFormatHybridEven byte = 0x06
+
+	// PubKeyIDHybridEven is the identifier prefix byte for a public key whose Y
+	// coordingate is odd when serialized according to the hybrid format per
+	// section 4.3.6 of [ANSI X9.62-1998].
+	//
+	// NOTE: This format makes little sense in practice an therefore this
+	// package will not produce public keys serialized in this format.  However,
+	// it will parse them since they exist in the wild.
+	PubKeyFormatHybridOdd byte = 0x07
 )
 
-// PublicKey provides facilities for efficiently working with secp256k1 private
+// PublicKey provides facilities for efficiently working with secp256k1 public
 // keys within this package and includes functions to serialize in both
 // uncompressed and compressed SEC (Standards for Efficient Cryptography)
 // formats.
 type PublicKey struct {
-	x *big.Int
-	y *big.Int
+	x FieldVal
+	y FieldVal
 }
 
-// X returns the x coordinate of the public key.
-func (p *PublicKey) X() *big.Int {
-	return p.x
-}
-
-// Y returns the y coordinate of the public key.
-func (p *PublicKey) Y() *big.Int {
-	return p.y
-}
-
-// NewPublicKey instantiates a new public key with the given X,Y coordinates.
-func NewPublicKey(x *big.Int, y *big.Int) *PublicKey {
-	return &PublicKey{
-		x: x,
-		y: y,
-	}
-}
-
-func isOdd(a *big.Int) bool {
-	return a.Bit(0) == 1
-}
-
-// decompressPoint decompresses a point on the given curve given the X point and
-// the solution to use.
-func decompressPoint(x *big.Int, ybit bool) (*big.Int, error) {
-	var fx, fy FieldVal
-	if overflow := fx.SetByteSlice(x.Bytes()); overflow {
-		str := "invalid public key: x >= field prime"
-		return nil, makeError(ErrPubKeyXTooBig, str)
-	}
-	if !DecompressY(&fx, ybit, &fy) {
-		str := fmt.Sprintf("invalid public key: x coordinate %v is not on the "+
-			"secp256k1 curve", fx)
-		return nil, makeError(ErrPubKeyNotOnCurve, str)
-	}
-	fy.Normalize()
-
-	return new(big.Int).SetBytes(fy.Bytes()[:]), nil
+// NewPublicKey instantiates a new public key with the given x and y
+// coordinates.  The provide values are truncated to 32 bytes and taken modulo
+// the secp256k1 field prime.
```

Comment:
I had originally split up the switch to the internal implementation using the field vals while leaving the public API using `big.Int`s and added that blurb due to that since some other things needed to change in order to be able to change the public API.  However, I ended up reordering the commits a bit and combining things and forgot to remove that.

Nice catch!

Reply:
Removed.